tools/analyzer_plugin/tool/doc_generation_utils/generate_contributor_type_index.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+
+import 'shared_html.dart';
+
+abstract class ContributorIndexer {
+  final String pageTitle;
+  final String bodySubtitle;
+  ContributorIndexer({
+    @required this.pageTitle,
+    @required this.bodySubtitle,
+  });
+
+  void generate(String filePath) {
+    var generated = _generate();
+    if (filePath != null) {
+      var outPath = '$filePath/index.html';
+      print('Writing to $outPath');
+      File(outPath).writeAsStringSync(generated);
```

Comment:
This fails if the parent dir doesn't exist
```suggestion
      File(outPath)
        ..parent.createSync(recursive: true)
        ..writeAsStringSync(generated);
```